lisa/transformers/kernel_installer.py

```diff
@@ -128,39 +128,34 @@ def _internal_run(self) -> Dict[str, Any]:
         self._log.info(f"installed kernel version: {installed_kernel_version}")
 
         # for ubuntu cvm kernel, there is no menuentry added into grub file
-        if (
-            hasattr(installer.runbook, "source")
-            and installer.runbook.source != "linux-image-azure-fde"
-        ):
-            posix = cast(Posix, node.os)
-            posix.replace_boot_kernel(installed_kernel_version)
-
-        if (
-            hasattr(installer.runbook, "source")
-            and installer.runbook.source == "linux-image-azure-fde"
-        ):
-            efi_files = node.execute(
-                "ls -t /usr/lib/linux/efi/kernel.efi-*-azure-cvm",
-                sudo=True,
-                shell=True,
-                expected_exit_code=0,
-                expected_exit_code_failure_message=(
-                    "fail to find kernel.efi file for kernel type linux-image-azure-fde"
-                ),
-            )
-            efi_file = efi_files.stdout.splitlines()[0]
-            node.execute(
-                (
-                    "cp /boot/efi/EFI/ubuntu/grubx64.efi "
-                    "/boot/efi/EFI/ubuntu/grubx64.efi.bak"
-                ),
-                sudo=True,
-            )
-            node.execute(
-                f"cp {efi_file} /boot/efi/EFI/ubuntu/grubx64.efi",
-                sudo=True,
-                shell=True,
-            )
+        if hasattr(installer.runbook, "source"):
+            if installer.runbook.source != "linux-image-azure-fde":
```

Comment:
It doesn't need to be changed, but it's better to use positive conditions by default, if there is special reason. Use `if installer.runbook.source == "linux-image-azure-fde":` first, it will be a little bit better to understand.